core/dtn7/src/routing/erouting/processing.rs

```diff
@@ -0,0 +1,255 @@
+        info!("Websocket connection closed because external routing is not enabled");
+        if let Err(err) = ws.close().await {
+            info!("Error while closing websocket: {}", err);
+        }
+        return;
+    }
+
+    let (tx, rx) = unbounded();
+
+    if CONNECTION.lock().unwrap().is_some() {
+        info!("Websocket connection closed because external routing agent is already connected");
+        if let Err(err) = ws.close().await {
+            info!("Error while closing websocket: {}", err);
+        }
+        return;
+    }
+
+    *CONNECTION.lock().unwrap() = Some(Connection { tx });
+
+    // Send initial states to the router
+    send_peer_state();
+    send_service_state();
+
+    let (outgoing, incoming) = ws.split();
+
+    let broadcast_incoming = incoming.try_for_each(|msg| {
+        trace!(
+            "Received a external routing message: {}",
+            msg.to_text().unwrap().trim()
+        );
+
+        let packet: serde_json::Result<Packet> = serde_json::from_str(msg.to_text().unwrap());
+
+        match packet {
+            Ok(packet) => match packet {
+                // When a SenderForBundleResponse is received we check if a response channel for that
+                // bundle id exists and send the response on that channel.
+                Packet::SenderForBundleResponse(packet) => {
+                    trace!(
+                        "sender_for_bundle response: {}",
+                        msg.to_text().unwrap().trim()
+                    );
+
+                    if let Some(tx) = RESPONSES
+                        .lock()
+                        .unwrap()
+                        .remove(packet.bp.to_string().as_str())
+                    {
+                        if let Err(_) = tx.send(Packet::SenderForBundleResponse(packet)) {
+                            error!("sender_for_bundle response could not be passed to channel")
+                        }
+                    } else {
+                        info!("sender_for_bundle no response channel available")
+                    }
+                }
+                // Add a service on packet
+                Packet::ServiceAdd(packet) => {
+                    info!(
+                        "adding service via erouting {}:{}",
+                        packet.tag, packet.service
+                    );
+
+                    service_add(packet.tag, packet.service);
+                }
+                _ => {}
+            },
+            Err(err) => {
+                info!("err decoding external routing packet: {}", err);
+            }
+        }
+
+        future::ok(())
+    });
+
+    let receive_from_others = rx.map(Ok).forward(outgoing);
+
+    pin_mut!(broadcast_incoming, receive_from_others);
+    future::select(broadcast_incoming, receive_from_others).await;
+
+    info!("External routing disconnected");
+    disconnect();
+}
+
+fn disconnect() {
+    if CONNECTION.lock().unwrap().is_some() {
```

Comment:
why is the check for some used? even if it is none, reassigning none does not hurt so the check is not needed